integration-tests/scaffolding/helpers.ts

```diff
@@ -73,3 +74,41 @@ export function log(...args: any[]) {
         console.log(...args);
     }
 }
+
+export async function createProviderKeysAndId(): Promise<[KeyringPair, u64]> {
+    let providerKeys = await createAndFundKeypair();
+    let createProviderMsaOp = ExtrinsicHelper.createMsa(providerKeys);
+    let providerId = new u64(ExtrinsicHelper.api.registry, 0)
+    await createProviderMsaOp.fundAndSend();
+    let createProviderOp = ExtrinsicHelper.createProvider(providerKeys, "PrivateProvider");
+    let [providerEvent] = await createProviderOp.fundAndSend();
+    if (providerEvent && ExtrinsicHelper.api.events.msa.ProviderCreated.is(providerEvent)) {
+        providerId = providerEvent.data.providerId;
+    }
+    return [providerKeys, providerId];
+}
+
+export async function createDelegatorAndDelegation(schemaId: u16, providerId: u64, providerKeys: KeyringPair): Promise<[KeyringPair, u64]> {
+    // Create a  delegator msa
+    let keys = await createAndFundKeypair();
+    let delegator_msa_id = new u64(ExtrinsicHelper.api.registry, 0);
+    const createMsa = ExtrinsicHelper.createMsa(keys);
+    await createMsa.fundOperation();
+    const [msaCreatedEvent, chainEvents] = await createMsa.signAndSend();
+    if (msaCreatedEvent && ExtrinsicHelper.api.events.msa.MsaCreated.is(msaCreatedEvent)) {
+        delegator_msa_id = msaCreatedEvent.data.msaId;
+    }
+
+    // Grant delegation to the provider
+    const payload = await generateDelegationPayload({
+        authorizedMsaId: providerId,
+        schemaIds: [schemaId],
+    });
+    const addProviderData = ExtrinsicHelper.api.registry.createType("PalletMsaAddProvider", payload);
+
+    const grantDelegationOp = ExtrinsicHelper.grantDelegation(keys, providerKeys, signPayloadSr25519(keys, addProviderData), payload);
+    await grantDelegationOp.fundOperation();
+    await grantDelegationOp.signAndSend();
+
+    return [keys, delegator_msa_id];
+}
```

Comment:
Some helper functions to create Provider and create Delegator with delegation activated